rust/phoenix-sdk/src/utils.rs

```diff
@@ -0,0 +1,94 @@
+use std::mem::size_of;
+
+use ellipsis_client::EllipsisClient;
+use phoenix::program::{dispatch_market, get_seat_address, MarketHeader};
+use phoenix_seat_manager::instruction_builders::{
+    create_claim_seat_instruction, create_evict_seat_instruction, EvictTraderAccountBackup,
+};
+use solana_program::{instruction::Instruction, pubkey::Pubkey};
+use spl_associated_token_account::{
+    get_associated_token_address, instruction::create_associated_token_account,
+};
+
+pub async fn create_ata_ix_if_needed(
+    client: &EllipsisClient,
+    payer: &Pubkey,
+    trader: &Pubkey,
+    mint: &Pubkey,
+) -> Option<Instruction> {
+    let ata_address = get_associated_token_address(trader, mint);
+    let ata_account = client.get_account(&ata_address).await;
+
+    if let Ok(ata_account) = ata_account {
+        if !ata_account.data.is_empty() {
+            return None;
+        }
+    }
+    Some(create_associated_token_account(
+        payer,
+        trader,
+        mint,
+        &spl_token::ID,
+    ))
+}
+
+// Check if seat already exists, if not, create seat instruction.
+// Check if the market trader state is full, if so, find a seat to evict and add the evict instruction.
+pub async fn create_claim_seat_ix_if_needed(
+    client: &EllipsisClient,
+    market_pubkey: &Pubkey,
+    trader: &Pubkey,
+) -> anyhow::Result<Vec<Instruction>> {
+    let seat_address = get_seat_address(market_pubkey, trader).0;
+    let seat_account = client.get_account(&seat_address).await;
```

Comment:
Should early return if seat already exists and is valid. I'm not sure if this happens right here or if you have to look it up in the trader tree

Reply:
cc @jarry-xiao

Reply:
If a seat is Approved I believe it is impossible in any state of the world (even outside of PSM) for it to not be on the book. This is because the seat needs to be NotApproved to evict and gets assigned a seat when it gets Approved.

I think it's fine to just terminate early here if we see Approved.